flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionID.java

```diff
@@ -44,7 +47,12 @@ public final class ResultPartitionID implements Serializable {
 
     @VisibleForTesting
     public ResultPartitionID() {
-        this(new IntermediateResultPartitionID(), new ExecutionAttemptID());
+        this(
+                new IntermediateResultPartitionID(),
+                new ExecutionAttemptID(
+                        new ExecutionGraphID(),
+                        new ExecutionVertexID(new JobVertexID(0, 0), 0),
+                        0));
```

Comment:
We could add a convenience constructor to avoid these changes.

Reply:
I'm thinking of adding a factory method `ExecutionAttemptID#createDummyExecutionAttemptID()` to explicitly show it is a dummy one to be used in production.
More details see my other [comment](https://github.com/apache/flink/pull/19747#discussion_r875516319).

Reply:
Ok. I've added an `ExecutionAttemptID#randomId()`.